### DISCUSSION TITLE
How to restart the docker container?

### DISCUSSION THREAD

**Discussion by jh20001** [zwave-js/zwave-js-ui#1995] [Q&A] [answered]
I installed it as volume (the first option) and set everything up. I added 2 devices and integrated them into automations within HA,

However, I updated Ubuntu and rebooted and noticed zwavejs2mqtt was no longer running. Portainer shows the container there, but I can't seem to figure out how to get the gateway running again.

**Comment by robertsLando** [comment]
Never used portainer btw if you start thghe volume using `docker compose up` it should persist across reboots

Maybe @blhoward2 can help here?

**Comment by blhoward2** [comment]
In portainer you'd go to containers, find the right one, go into it, then click start. If it's not starting click logs to see why.

**Comment by jh20001** [comment]
Ok, I just realized I was looking from the dashboard and was clicking on images. It isn't listed under containers. It is listed under volumes and images, and neither of those screens give an option to start it. It does show it in volumes though (zwavejs2mqtt Unused | - | local | /var/lib/docker/volumes/zwavejs2mqtt/_data)

I initially installed it via the first of the three options (volumes), which seemed to work fine. Is there something I should have added within the terminal that wasn't listed in the walkthrough? during install?

**Reply by blhoward2** [reply]
A volume is just a docker compose. If there is no corresponding container then you just need to deploy it. Go into the volume, go to editor, then click update. There is likely also a deploy button in there somewhere.

**Comment by jh20001** [comment, marked as answer]
Sadly, there is no editor, deploy, or update option available within the volume page of Portainer for this. It might need to be done via Terminal. Maybe since I didn't create any of this through Portainer. I just installed it in the past to view everything that is installed/running.  I used the "docker run --rm -it -p 8091:8091 -p 3000:3000 --device=/dev/serial/by-id/insert_stick_reference_here:/dev/zwave \
--mount source=zwavejs2mqtt,target=/usr/src/app/store zwavejs/zwavejs2mqtt:latest" code (with the stick/id information properly filled out. Then configured it once I pulled it up via browser. Everything was working fine will I rebooted. 

![image](https://user-images.githubusercontent.com/6210275/143666823-882572dc-a7aa-4b87-9464-3503a4bcca8d.png)

**Reply by blhoward2** [reply]
Ah, I was thinking of Stacks. The volume is just where the data is stored...think if it like a disk drive. It has nothing to do with the container itself. If it isn't showing under Stacks and no container is running, then of course it won't start. Copy our template docker compose into a stack and then deploy it (referencing your volume for the store). Or restart it from the command line.

**Comment by jh20001** [comment]
Am I using the run as a service template? *(https://zwave-js.github.io/zwavejs2mqtt/#/getting-started/docker?id=run-as-a-service)

If so, it produces an error about no declaration. 

![image](https://user-images.githubusercontent.com/6210275/143667613-a60b6c4c-d0c9-4b49-a116-301e0791b885.png)

**Reply by blhoward2** [reply]
Yes. That's because you've renamed it up top but not at the bottom where the volume is configured. You have zwavejs2mqtt in one place and zwavejs-config in another.

**Comment by jh20001** [comment]
Good catch! Sorry. I've been up for close to 20 hours now thanks to long work hours. I changed that and I was able to get into the control panel. I may be ready to go now. THANK YOU for helping out. I do appreciate it a lot :)
